src/main/java/org/springframework/data/elasticsearch/core/mapping/IndexCoordinatesResolver.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core.mapping;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.expression.BeanFactoryAccessor;
+import org.springframework.context.expression.BeanFactoryResolver;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ParserContext;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.util.Assert;
+
+/**
+ * A class to resolve index names (or generally Strings) by using an ApplicationContext aware SpELExpressionParser. The
+ * index names to resolve must be SpEL template expressions.
+ *
+ * @author Peter-Josef Meisch
+ */
+public class IndexCoordinatesResolver implements ApplicationContextAware {
+
+	private final StandardEvaluationContext context;
```

Comment:
SpEL should make use of Spring Data's `EvaluationContext` resolution which is available from within `BasicPersistentEntity`. What's the need for introducing a wrapper for SpEL evaluation?

In other modules, we do not leak outside of the entity/property the fact that a SpEL expression is used for the collection/table name.

Reply:
We do not always have an entity for example in the IndexOperations, for example when `IndexOperations ElasticsearchOperations.indexOps(IndexCoordinates index)` is used.

BTW, I didn't know that there is a EvaluationContext available, the code that I use for the SpEL evaluation was basically moved from  `SimpleElasticsearchPersistenEntity`

Reply:
I'd like to reuse the `EvaluationContextProvider` from `MappingContext`, but it is private